polaris/utils/types.py

```diff
@@ -125,6 +125,13 @@ class HubOwner(BaseModel):
     def __str__(self):
         return self.slug
 
+    @staticmethod
```

Comment:
**Question:** Should this be a `@classmethod` instead?

Reply:
I mean, you could implement it as a class method and do:
```python
@classmethod
def normalize(cls, owner: Union[str, "HubOwner"]) -> "HubOwner":
    """
    Normalize a string or `HubOwner` instance to a `HubOwner` instance.
    """
    return owner if isinstance(owner, cls) else cls(slug=owner)
```

Feels slightly less clear to me, since we need to the class as a type in the signature anyway.

Reply:
I think this may be a misunderstanding on my end. I actually thought a `@classmethod` _had to_ return an object of that class. I do thus strongly associate class methods with factory-like methods, but if that association is less for you I'm okay with keeping it this way!

Reply:
I see! No worries. Class methods can be useful when you have inheritance. Children classes inherit the method and will get the their class passed in. So the parent class can implement generic logic that could work for all derived children.

Typing becomes a bit more complex. Before Python 3.11, you have to do something like this:
```python
from typing import Type, TypeVar

T = TypeVar('T', bound='Base')

class Base:
    def __init__(self, value: int) -> None:
        self.value = value

    @classmethod
    def create(cls: Type[T], value: int) -> T:
        return cls(value)

class Child(Base):
    def method_not_present_on_base(self) -> None:
        print("This method is only in Child")

base = Base.create(1)
child = Child.create(2)
child.method_not_present_on_base()  # This works correctly
```

Python 3.11 introduces a [Self](https://docs.python.org/3/library/typing.html#typing.Self) type that makes it cleaner:
```python
from typing import Self

class Base:
    def __init__(self, value: int) -> None:
        self.value = value

    @classmethod
    def create(cls) -> Self:
        return cls(42)

class Child(Base):
    def method_not_present_on_base(self) -> None:
        print("This method is only in Child")

base = Base.create()
child = Child.create()
child.method_not_present_on_base()  # This works correctly
```